### PR TITLE
Add index for 1.0 migration guide

### PR DESCRIPTION
**Pull request by frankharkins**
* Adds a brief 1.0 migration guide index
* Adds all pages to the TOC

### CODE REVIEW

File: docs/api/migration-guides/index.mdx [hunk 1]
```diff
@@ -8,6 +8,9 @@ description: Migrate to using the newest from Qiskit and Qiskit Runtime
 
 We've prepared various migration guides to help you more effectively use Qiskit&reg; and Qiskit Runtime:
 
+* Migrate to Qiskit 1.0
+   * [Qiskit 1.0 packaging changes](/api/docs/migration-guides/qiskit-1.0-installation)
```
Comment (abbycross) [inline docs/api/migration-guides/index.mdx:12]:
```suggestion
   * [Qiskit 1.0 packaging changes](/api/migration-guides/qiskit-1.0-installation)
```

File: docs/api/migration-guides/index.mdx [hunk 2]
```diff
@@ -8,6 +8,9 @@ description: Migrate to using the newest from Qiskit and Qiskit Runtime
 
 We've prepared various migration guides to help you more effectively use Qiskit&reg; and Qiskit Runtime:
```
Comment (abbycross) [inline docs/api/migration-guides/index.mdx:9]:
```suggestion
These migration guides help you more effectively use Qiskit&reg; and Qiskit Runtime:
```

Comment (abbycross) [reply docs/api/migration-guides/index.mdx:9]: (removing "various" to make it less vague, and rewording to avoid the "we")

**Review by abbycross — APPROVED** [review]
A couple small suggestions, otherwise looks good to me!

File: docs/api/migration-guides/qiskit-1.0.mdx [hunk 3]
```diff
@@ -1,685 +1,15 @@
...
-description: Adapt to changes in installing and depending on Qiskit 1.0
+title: Qiskit 1.0 migration guide
+description: Update your project to use Qiskit 1.0
 ---
 
-# Qiskit 1.0 packaging changes
+# Qiskit 1.0 migration guide
 
-Qiskit&reg; 1.0 uses a different packaging structure than previous Qiskit versions and might cause problems in environments that use packages that are not ready for Qiskit 1.0.
+Qiskit 1.0 brings new stability guarantees and performance improvements, but it
```
Comment (abbycross) [inline docs/api/migration-guides/qiskit-1.0.mdx:8]:
```suggestion
Qiskit&reg; 1.0 brings new stability guarantees and performance improvements, but it
```

File: docs/api/migration-guides/qiskit-1.0.mdx [hunk 4]
```diff
@@ -1,685 +1,15 @@
...
-
-If you are a Qiskit developer, it's possible that you have old `qiskit.egg-info` or `qiskit-terra.egg-info` (or `*.dist-info`) directories present on your meta path (see `sys.meta_path`), left over from old editable installations.
-In particular, check your working directory for any `*.egg-info` and `*.dist-info` directories.
-If they're in the root of one of your checked out repositories, you can delete them. The worst that can happen is you might need to `pip install -e .` again, and even that is unlikely, because these are typically just part of the `setuptools` build process that doesn't get cleaned up.
-
-If the above information does not help you and you are 100% sure that your environment is sound (or you are deliberately trying to test a broken environment):
-
-1. [Create an issue in Qiskit](https://github.com/Qiskit/qiskit/issues/new/choose) explaining how this happened and why you are sure the environment is correct so we can fix it.
-2. You can suppress the exception by setting the environment variable `QISKIT_SUPPRESS_1_0_IMPORT_ERROR=1`.
+Read more about the benefits of Qiskit 1.0 on the [IBM blog](https://www.ibm.com/quantum/blog/qiskit-1-launch).
```
Comment (abbycross) [inline docs/api/migration-guides/qiskit-1.0.mdx:15]:
```suggestion
Read more about the benefits of Qiskit 1.0 on the [IBM&reg; blog](https://www.ibm.com/quantum/blog/qiskit-1-launch).
```

File: docs/api/migration-guides/index.mdx [hunk 5]
```diff
@@ -8,6 +8,9 @@ description: Migrate to using the newest from Qiskit and Qiskit Runtime
...
 
+* Migrate to Qiskit 1.0
+   * [Qiskit 1.0 packaging changes](/api/docs/migration-guides/qiskit-1.0-installation)
+   * [Qiskit 1.0 feature changes](/api/docs/migration-guides/qiskit-1.0-installation)
```
[the remark this answers is not in the document]
Comment (frankharkins) [reply docs/api/migration-guides/index.mdx:13]: Ooops thank you

**Review by Eric-Arellano — APPROVED** [review]

**Pull request merged**